array_api_strict/tests/test_creation_functions.py

```diff
@@ -236,3 +236,9 @@ def from_dlpack_2023_12(api_version):
     pytest.raises(exception, lambda: from_dlpack(capsule, copy=False))
     pytest.raises(exception, lambda: from_dlpack(capsule, copy=True))
     pytest.raises(exception, lambda: from_dlpack(capsule, copy=None))
+
+
+def test_from_dlpack_default_device():
+    x = asarray([1, 2, 3])
+    y = from_dlpack(x)
+    assert x.device is y.device
```

Comment:
I think this should be
```suggestion
    assert x.device == y.device
```
because you can create `Device()` instances as you want and `Device` isn't a singleton or something like that. So `xp.Device(device="device2") is xp.Device("device2")` is `False`.

I think in this case the test passes because the instance for the default device is reused, but that seems like a "happy accident"/implementation detail?

Tl;DR: there is AFAIK no downside to `==` and it will always work